llvm/include/llvm/ExecutionEngine/JITLink/systemz.h

```diff
@@ -0,0 +1,958 @@
+      return makeTargetOutOfRangeError(G, B, E);
+    write32be(FixupPtr, Value);
+    break;
+  }
+  case RequestGOTAndTransformToDelta32GOTBasedbl: {
+    assert(GOTSymbol && "No GOT section symbol");
+    int64_t Value = (GOTBase + A - P);
+    if (!LLVM_UNLIKELY(isInt<33>(Value)))
+      return makeTargetOutOfRangeError(G, B, E);
+    if (!LLVM_UNLIKELY(isAlignmentCorrect(Value, 2)))
+      return makeAlignmentError(FixupAddress, Value, 2, E);
+    write32be(FixupPtr, Value >> 1);
+    break;
+  }
+  case DeltaPLT32dbl: {
+    int64_t Value = (S + A - P);
+    if (!LLVM_UNLIKELY(isInt<33>(Value)))
+      return makeTargetOutOfRangeError(G, B, E);
+    if (!LLVM_UNLIKELY(isAlignmentCorrect(Value, 2)))
+      return makeAlignmentError(FixupAddress, Value, 2, E);
+    write32be(FixupPtr, Value >> 1);
+    break;
+  }
+  case DeltaPLT24dbl: {
+    int64_t Value = (S + A - P);
+    if (!LLVM_UNLIKELY(isInt<25>(Value)))
+      return makeTargetOutOfRangeError(G, B, E);
+    FixupPtr[0] = Value >> 17;
+    FixupPtr[1] = Value >> 9;
+    FixupPtr[2] = Value >> 1;
+    break;
+  }
+  case DeltaPLT16dbl: {
+    int64_t Value = (S + A - P);
+    if (!LLVM_UNLIKELY(isInt<17>(Value)))
+      return makeTargetOutOfRangeError(G, B, E);
+    if (!LLVM_UNLIKELY(isAlignmentCorrect(Value, 2)))
+      return makeAlignmentError(FixupAddress, Value, 2, E);
+    write16be(FixupPtr, Value >> 1);
+    break;
+  }
+  case DeltaPLT12dbl: {
+    int64_t Value = (S + A - P);
+    if (!LLVM_UNLIKELY(isInt<13>(Value)))
+      return makeTargetOutOfRangeError(G, B, E);
+    if (!LLVM_UNLIKELY(isAlignmentCorrect(Value, 2)))
+      return makeAlignmentError(FixupAddress, Value, 2, E);
+    write16be(FixupPtr,
+              (read16be(FixupPtr) & 0xF000) | ((Value >> 1) & 0x0FFF));
+    break;
+  }
+  case DeltaPLT64: {
+    int64_t Value = (S + A - P);
+    write64be(FixupPtr, Value);
+    break;
+  }
+  case DeltaPLT32: {
+    int64_t Value = (S + A - P);
+    if (!LLVM_UNLIKELY(isInt<32>(Value)))
+      return makeTargetOutOfRangeError(G, B, E);
+    write32be(FixupPtr, Value);
+    break;
+  }
+  case DeltaPLT64FromGOT: {
+    assert(GOTSymbol && "No GOT section symbol");
+    int64_t Value = (S + A - GOTBase);
+    write64be(FixupPtr, Value);
+    break;
+  }
+  case DeltaPLT32FromGOT: {
+    assert(GOTSymbol && "No GOT section symbol");
+    int64_t Value = (S + A - GOTBase);
+    if (!LLVM_UNLIKELY(isInt<32>(Value)))
+      return makeTargetOutOfRangeError(G, B, E);
+    write32be(FixupPtr, Value);
+    break;
+  }
+  case DeltaPLT16FromGOT: {
+    assert(GOTSymbol && "No GOT section symbol");
+    int64_t Value = (S + A - GOTBase);
+    if (!LLVM_UNLIKELY(isInt<16>(Value)))
+      return makeTargetOutOfRangeError(G, B, E);
+    write16be(FixupPtr, Value);
+    break;
+  }
+  case Branch32dblToStub: {
+    int64_t Value = (S + A - P);
+    if (!LLVM_UNLIKELY(isInt<33>(Value)))
+      return makeTargetOutOfRangeError(G, B, E);
+    char *AddrToPatch = FixupPtr + 2;
```

Comment:
As mentioned above, I don't think we need this at all.  This should simply be a `Delta32dbl` with an offset of 2.